### PR TITLE
Add App UI catalog

### PR DESCRIPTION
**Pull request by wmontwe**
This changes the `core:ui:compose:demo` module to `app-ui-catalog`. It is a catalog of all the components in the Thunderbird design system and allows switching between the themes and light/dark mode.

### CODE REVIEW

File: app-ui-catalog/src/main/java/app/k9mail/ui/catalog/CatalogTheme.kt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package app.k9mail.ui.catalog
+
+enum class CatalogTheme(
+    private val displayName: String,
+) {
+    K_9("K-9"),
```
Comment (cketti) [inline app-ui-catalog/src/main/java/app/k9mail/ui/catalog/CatalogTheme.kt:6]: `K_9` looks weird. Can we make it `K9`?

File: app-ui-catalog/src/main/java/app/k9mail/ui/catalog/CatalogContent.kt [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    catalogTheme: CatalogTheme,
+    catalogThemeVariant: CatalogThemeVariant,
+    onThemeChange: () -> Unit,
+    onThemeVariantChange: () -> Unit,
+    contentPadding: PaddingValues,
+    modifier: Modifier = Modifier,
+) {
+    Surface {
+        LazyVerticalGrid(
+            columns = GridCells.Adaptive(300.dp),
```
Comment (cketti) [inline app-ui-catalog/src/main/java/app/k9mail/ui/catalog/CatalogContent.kt:33]: To me it doesn't feel or look great to have more than one column. I'd only set a maximum width for the content area.

File: app-ui-catalog/src/main/java/app/k9mail/ui/catalog/items/ButtonItems.kt [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import app.k9mail.core.ui.compose.designsystem.atom.button.Button
+import app.k9mail.core.ui.compose.designsystem.atom.button.ButtonOutlined
+import app.k9mail.core.ui.compose.designsystem.atom.button.ButtonText
+import app.k9mail.core.ui.compose.theme.MainTheme
+
+fun LazyGridScope.buttonItems() {
+    sectionHeaderItem(text = "Buttons")
+    sectionSubtitleItem(text = "Contained")
+    item {
+        LazyRow(
```
Comment (cketti) [inline app-ui-catalog/src/main/java/app/k9mail/ui/catalog/items/ButtonItems.kt:15]: Wouldn't `Row` do here?

**Review by cketti — APPROVED** [review]

File: app-ui-catalog/src/main/java/app/k9mail/ui/catalog/CatalogContent.kt (hunk 2 above)
Comment (wmontwe) [reply app-ui-catalog/src/main/java/app/k9mail/ui/catalog/CatalogContent.kt:33]: Setting a maximum width requires additional work, as it doesn't work out of the box.

Comment (wmontwe) [reply app-ui-catalog/src/main/java/app/k9mail/ui/catalog/CatalogContent.kt:33]: I'll add another approach with the onboarding screen and could apply that to the catalog too.

**Pull request merged**